### PR TITLE
Cleanup changelog references in Introduction.md

### PR DESCRIPTION
**Pull request by stwrt**
#### 21704b5f7c44584a831986498dbcc8e0f85affd7
<pre>
Cleanup changelog references in Introduction.md
<a href="https://bugs.webkit.org/show_bug.cgi?id=242433">https://bugs.webkit.org/show_bug.cgi?id=242433</a>

Reviewed by Jonathan Bedard and Ryosuke Niwa.

Update Introduction.md to remove references to changelogs, and
enhance the &quot;commit messages&quot; section.

* Introduction.md:

Canonical link: <a href="https://commits.webkit.org/252241@main">https://commits.webkit.org/252241@main</a>
</pre>

### CODE REVIEW

File: Introduction.md [hunk 1]
```diff
@@ -299,34 +299,10 @@ Running `git-webkit setup` configures your git hooks to properly generate commit
 The first line contains the date, your full name, and your email address.
 Use this to write up a brief summary of the changes you’ve made.
 Don’t worry about the “Reviewed by NOBODY (OOPS!)” line, the person landing your patch will fill this in.
-There is one ChangeLog per top-level directory.
 
-A typical change log entry before being submitted to [bugs.webkit.org](https://bugs.webkit.org/) looks like this:
```
Comment (JonWBedard) [inline Introduction.md:304]: Seems like we should provide a commit message example here.

Comment (stwrt) [reply Introduction.md:304]: Sure, I can add that

File: Introduction.md [hunk 2]
```diff
@@ -296,37 +296,46 @@ It’s in your interest to wait for the Early Warning System to fully build and
...
+Add WKNavigation SPI to check if the navigation had a user interaction.
+
+Also make sure that the value is correct for same-document navigations. In
+particular, navigations via history.pushState() / history.popState() / history.replaceState()
+would previously reuse the last WKNavigation object as they wouldn't get their
+own navigation object. I have made it so that these navigations now get their
+own navigation object and that the _isUserInitiated flag is correct for this
+navigation object.
+
+* Source/WebKit/UIProcess/API/Cocoa/WKNavigation.mm:
```
Comment (rniwa) [inline Introduction.md]: We want to use a commit message with per-function comments.

Comment (stwrt) [reply Introduction.md]: Found a commit with a couple of per-function comments.

File: Introduction.md [hunk 3]
```diff
@@ -296,37 +296,47 @@ It’s in your interest to wait for the Early Warning System to fully build and
 Commit messages serve as change logs, providing historical documentation for all changes to the WebKit project.
 Running `git-webkit setup` configures your git hooks to properly generate commit messages.
 
-The first line contains the date, your full name, and your email address.
-Use this to write up a brief summary of the changes you’ve made.
-Don’t worry about the “Reviewed by NOBODY (OOPS!)” line, the person landing your patch will fill this in.
-There is one ChangeLog per top-level directory.
-
-A typical change log entry before being submitted to [bugs.webkit.org](https://bugs.webkit.org/) looks like this:
+The first line shall contain a short description of the commit message (this should be the same as the Summary field in Bugzilla). On the next line, enter the Bugzilla URL. Below the "Reviewed by" line, enter a detailed description of your changes.
```
Comment (rniwa) [inline Introduction.md]: Can we insert more hard line breaks? This is a really long line.

**Review by rniwa — APPROVED** [review]

File: Introduction.md (hunk 3 above)
Comment (stwrt) [reply Introduction.md]: Sure 😀

Comment (stwrt) [reply Introduction.md]: Added some notes about commenting here too.

**Review by JonWBedard — APPROVED** [review]

**Pull request merged**